include/ideep/attributes.hpp

```diff
@@ -352,14 +352,12 @@ struct attr_t : public dnnl::primitive_attr {
     if (l_po.len() != r_po.len() ||
         get_output_scales() != rhs.get_output_scales() ||
         get_fpmath_mode() != rhs.get_fpmath_mode() ||
-        get_scratchpad_mode() != rhs.get_scratchpad_mode()) {
```

Comment:
This change breaks the semantics of general equality of `attr_t` type and sounds very dangerous. Can we alter the caller to do the comparison without considering scatchpad_mode? Maybe we can add another method like `equals_except_scratchpad_mode` to be invoked by the callers? Thoughts?

Reply:
By intuition operator== should compare everything. But here it's used for guard of prepacked weight and cached primitive in IPEX. It is not used in stock or internal PyTorch.
I think in IPEX the semantics are not very strict. They just check what they care. In current IPEX (without third_party/ideep), the operator== only check post ops, nothing else (https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-cpu/blob/3a249716baf967a10218d6907ac2c0a3ffc963ab/intel_extension_for_pytorch/csrc/cpu/ideep/ideep/attributes.hpp#L342).
In oneDNN, there is no official way to check if two attr's are exactly equal.
I think the best way is to remove this operator and use a util function to check parameters of interest in `attr` in IPEX. That would be out of scope of ideep.
Do you think we should remove it? @XiaobingSuper please also comment. Thanks.

Reply:
> I think the best way is to remove this operator and use a util function to check parameters of interest in attr in IPEX. That would be out of scope of ideep.

We can define the util function as a method of `attr_t` which only checks post-ops to keep semantics in IPEX. IPEX also needs to be changed.

Reply:
Remove this operator is ok, yes, we can define a util function to check it.

Reply:
OK I removed `operator==` and added `has_same_postop_as`